packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -77,37 +81,29 @@ contract SgxVerifier is EssentialContract, IVerifier {
         ids = _addInstances(_instances);
     }
 
-    /// @notice Adds SGX instances to the registry by another SGX instance.
-    /// @param id The id of the SGX instance who is adding new members.
-    /// @param newInstance The new address of this instance.
-    /// @param extraInstances The address array of SGX instances.
-    /// @param signature The signature proving authenticity.
-    /// @return ids The respective instanceId array per addresses.
-    function addInstances(
-        uint256 id,
-        address newInstance,
-        address[] calldata extraInstances,
-        bytes calldata signature
-    )
+    /// @notice Adds an SGX instance after the attestation is verified
+    /// @param attestation The parsed attestation quote.
+    /// @return id The respective instanceId
+    function registerInstance(V3Struct.ParsedV3QuoteStruct calldata attestation)
         external
-        returns (uint256[] memory ids)
+        returns (uint256)
     {
-        address taikoL1 = resolve("taiko", false);
-        bytes32 signedHash = keccak256(
-            abi.encode(
-                "ADD_INSTANCES",
-                ITaikoL1(taikoL1).getConfig().chainId,
-                address(this),
-                newInstance,
-                extraInstances
-            )
-        );
-        address oldInstance = ECDSA.recover(signedHash, signature);
-        if (!_isInstanceValid(id, oldInstance)) revert SGX_INVALID_INSTANCE();
+        address automataDcapAttestation = (resolve("automata_dcap_attestation", true));
 
-        _replaceInstance(id, oldInstance, newInstance);
+        if (automataDcapAttestation == address(0)) {
+            return type(uint256).max;
+        }
```

Comment:
Why doesn't this simply revert?

Reply:
Reason is: we might not support (for whatever reason) the on-chain attestastion, so it can be address(0) - and it will be backward compatible with previous versions.

Reply:
This code still needs updating to make it backward compatible or no? Could be that I'm missing something.

Reply:
We allow not having set a "dcap_attesation" in the address manager. If we add it, we enable the feature. If we dont add it, feature is not enabled.
Backward compatible with tests (too) :)

Reply:
So instead of reverting it returns an invalid ID instead? Reverting seems nicer so it's more clear it's disabled I think, but fair enough.